src/domain/runners/SetGroupMembershipBlockRunner.ts

```diff
@@ -0,0 +1,34 @@
+import {evaluateToBool, IBlockExit, IContext, IRichCursor, ISetGroupMembershipBlock, IBlockRunner, ValidationException} from '../..'
+
+const EXIT_SUCCESS = 0
+
+/**
+ * Adds or removes a group from the contact.
+ */
+export class SetGroupMembershipBlockRunner implements IBlockRunner {
+  constructor(public block: ISetGroupMembershipBlock, public context: IContext) {}
+
+  async initialize(): Promise<undefined> {
+    return undefined
+  }
+
+  async run(_cursor: IRichCursor): Promise<IBlockExit> {
+    const {exits} = this.block
+    const {contact, groups} = this.context
+    const {groupKey, isMember} = this.block.config
+
+    const group = groups.find(group => group.groupKey === groupKey)
+
+    if (group == null) {
+      throw new ValidationException(`Cannot add contact to non-existent group ${groupKey}`)
```

Comment:
Unsure about this.
Should the flowrunner care if a group isn't in the flow context?
Do we treat groups as _entities_ or just as objects that reduce down to a string?

Reply:
I think so, as this is a Data issue. It's better to throw Errors with dat issues, than it is to continue on like there's no problem at all, as that bad data propagates throughout the entire system, and other connected systems.